### PR TITLE
add eventhandler framework proposal

### PR DESCRIPTION
**Pull request by sivanzcw**

### CODE REVIEW

**Comment by william-wang** [comment]
@sivanzcw Thanks for your proposal. I will have a look.

**Comment by shinytang6** [comment]
Intersting, will be some share in the weekly meeting? @sivanzcw

File: docs/design/event-handler-framework.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+To access volcano as one framework to a two-level scheduler, the volcano needs to provide access for external modifications 
+to its cached resource objects.
+
+## Goals
+
+- volcano provides an entrance for external modification of its cached resource objects
+
+## Design Details
+
+ ![event-handler-arch](./images/event-handler-arch.png)
```
Comment (william-wang) [inline docs/design/event-handler-framework.md:45]: The image file is not avaliable.

File: docs/design/event-handler-framework.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+### EventHandler Factory
+
+- define an interface, including object event processing function set
+- provide a factory to realize the object event processing function
+- provide a factory registration mechanism
+
+```go
+// Interface containers all involved object event processing interfaces.
+type Interface interface {
+   AddPod(obj interface{})
```
Comment (william-wang) [inline docs/design/event-handler-framework.md:128]: what's the relation with AddPod with existing AddPod() in event_handlers.go

File: docs/design/event-handler-framework.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+}
+
+func (eh *defaultEventHandler) AddPod(obj interface{}) {
+   pod, ok := obj.(*v1.Pod)
+   if !ok {
+      klog.Errorf("Cannot convert to *v1.Pod: %v", obj)
+      return
+   }
+
+   sc := eh.cache
```
Comment (william-wang) [inline docs/design/event-handler-framework.md:183]: Is there any change against the package structure?  new added eventhandler and schedulerCache are still in the same package?

Comment (sivanzcw) [reply docs/design/event-handler-framework.md:183]: The default eventhandler is implemented in the original package.

**Review by william-wang — APPROVED** [review]
/lgtm

**Pull request merged**